tests/qepsi4/_psi4_test.py

```diff
@@ -48,199 +81,237 @@ def create_molecule(geometry, charge, mult):
+
+    @pytest.mark.parametrize(
+        "n_active_extract,n_occupied_extract,freeze_core_extract,expected",
+        [
+            (2, None, False, (0, 2, 5)),
+            (None, None, False, (0, 7, 0)),
+            (None, None, True, (1, 6, 0)),
+            (None, 3, True, (2, 5, 0)),
+            (4, 3, False, (2, 4, 1)),
+        ],
+    )
+    def test_active_space(
+        self,
+        water_mol_and_wfn,
+        n_active_extract,
+        n_occupied_extract,
+        freeze_core_extract,
+        expected,
+    ):
+        ndocc, nact, nvir = select_active_space(
+            *water_mol_and_wfn,
+            n_active_extract=n_active_extract,
+            n_occupied_extract=n_occupied_extract,
+            freeze_core_extract=freeze_core_extract,
+        )
 
-    assert math.isclose(energy, results["energy"])
+        assert (ndocc, nact, nvir) == expected
+
+
+class TestRunPsi4:
+    @pytest.mark.parametrize(
+        "psi4_config,jw_wagner_particle_num,expected_tuple",
+        [
+            (
+                Psi4Config(hydrogen_geometry, 1, options={"scf_type": "pk"}),
+                2,
+                ExpectedTuple(-0.8543376267387818, 1, 1, 2, 0, 1, 2),
+            ),
+            (
+                Psi4Config(hydrogen_geometry, 1, 0, options={"scf_type": "pk"}),
+                0,
+                ExpectedTuple(-0.8543376267387818, 1, 1, 2, 1, 0, 2),
+            ),
+            (
+                Psi4Config(hydrogen_geometry),
+                2,
+                ExpectedTuple(
+                    -0.8544322638069642, 1, 1, 2, 0, 0, 4, -0.9714266904819895
+                ),
+            ),
+            (
+                Psi4Config(dilithium_geometry, 4, 2,),
+                4,
+                ExpectedTuple(exp_n_qubits=8, extra_energy_check=-14.654620243980217),
+            ),
+            (
+                Psi4Config(hydrogen_geometry, method="fci", save_rdms=True),
+                2,
+                ExpectedTuple(None, 1, 1, 2, 0, 0, 4),
+            ),
+            (
+                Psi4Config(dilithium_geometry, 4, 1, method="fci", save_rdms=True),
+                2,
+                ExpectedTuple(exp_n_qubits=8, exp_one_body_tensor_shape=8),
+            ),
+            (
+                Psi4Config(
+                    dilithium_geometry,
+                    freeze_core=True,
+                    freeze_core_extract=True,
+                    method="fci",
+                    save_rdms=True,
+                ),
+                None,
+                ExpectedTuple(exp_n_qubits=16, exp_one_body_tensor_shape=16),
+            ),
+            (
+                Psi4Config(
+                    dilithium_geometry, freeze_core=True, freeze_core_extract=True
+                ),
+                None,
+                ExpectedTuple(exp_n_qubits=16),
+            ),
+        ],
+    )
+    def test_run_psi4(
+        self,
+        psi4_config: Psi4Config,
+        jw_wagner_particle_num: int,
```

Comment:
`jw` stands for Jordan-Wigner, so we can remove `wagner` from the variable name.